app/services/notification/ScheduledNotificationRunner.scala

```diff
@@ -0,0 +1,49 @@
+package services.notification
+
+import com.amazonaws.services.simpleemail.model._
+import config.AMIableConfig
+import models.{Attempt, Instance, Owner, SSA}
+import play.api.{Configuration, Logger}
+import prism.{Prism, PrismLogic}
+
+import scala.concurrent.ExecutionContext
+
+object ScheduledNotificationRunner {
+  // Message id used if an owner doesn't have any old instances, or doesn't own any instances at all
+  val MessageNotSent = "0"
+
+  def run(mailClient: AWSMailClient)(implicit config: AMIableConfig, ec: ExecutionContext): Attempt[List[String]] = {
+    for {
+      instancesWithAmis <- Prism.instancesWithAmis(SSA(stage = Some("PROD")))
+      oldInstances = PrismLogic.oldInstances(instancesWithAmis)
+      owners <- Prism.getOwners
+      mailIds <- Attempt.traverse(owners) { owner =>
+        val ownerOldInstances = instancesForOwner(owner, oldInstances)
```

Comment:
Just noticed from the PR description that this is your intention, so perhaps this isn't relevant.